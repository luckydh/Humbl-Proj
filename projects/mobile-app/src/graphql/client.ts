import { ApolloClient, createHttpLink, InMemoryCache, from, ApolloLink } from "@apollo/client";
import { setContext } from "@apollo/client/link/context";
import { onError } from "@apollo/client/link/error";
import "firebase/auth";
import { Storage } from "@capacitor/storage";
import { trackEvent } from "utils/analytics/Segment";
import EVENTS from "utils/analytics/AnalyticEvents";
import { setMinAppVersion, currAppVersion, minAppVersionVar as currMinAppVersion } from "utils/update/update";
import { captureException } from "ErrorLogger";
import { GraphQLError } from "graphql";
import firebase from "../Firebase";

const httpLink = createHttpLink({
  uri: process.env.REACT_APP_HUMBL_DATA_URL,
  // credentials: 'same-origin',
});

const error = onError(({ operation, networkError, graphQLErrors, response }) => {
  trackEvent(EVENTS.API_ERROR, {
    operation,
    networkError,
    graphQLErrors,
  });
  if (networkError) {
    captureException(networkError);
  }

  if (graphQLErrors) {
    graphQLErrors.forEach((e) => {
      const sentryError = new Error();
      const name = e.extensions?.code ?? e.message;
      sentryError.name = name;
      sentryError.message = e.message;
      captureException(sentryError, generateGraphQLErrorMeta(e));
    });
  }

  if (response?.errors) {
    // eslint-disable-next-line no-param-reassign
    response.errors = response.errors.map((err) => ({
      ...err,
      humblErrorCode: err.extensions?.code,
    }));
  }
});

function generateGraphQLErrorMeta(graphqlError: GraphQLError) {
  return Object.entries(graphqlError).reduce((curr, next) => {
    const [k, v] = next;
    // Ignore message since we use that as the main error text anyways.
    if (k !== "message") {
      try {
        curr[k] = JSON.stringify(v);
      } catch (e) {
        captureException(e);
      }
    }

    return curr;
  }, {} as Record<string, string>);
}

const authLink = setContext(async (_, { headers }) => {
  // get the authentication token from firebase if it exists
  const token = await firebase.auth().currentUser?.getIdToken();
  // return the headers to the context so httpLink can read them
  Storage.migrate();
  const accountId = await Storage.get({ key: "defaultAccountId" });
  const headerObject = {
    ...headers,
    authorization: token ? `Bearer ${token}` : "",
    version: currAppVersion,
  };
  // If we have a default account in local storage to work with, send it along
  if (accountId.value) {
    headerObject.account = accountId.value;
  }
  return {
    headers: headerObject,
  };
});

const responseLink = new ApolloLink((operation, forward) => forward(operation).map((response) => {
    const context = operation.getContext();
    const minAppVersion: string = context?.response?.headers?.get("minimum-app-version") ?? "";

    if (minAppVersion !== currMinAppVersion()) {
      setMinAppVersion(minAppVersion);
    }

    return response;
  }));

const client = new ApolloClient({
  cache: new InMemoryCache(),
  uri: process.env.REACT_APP_HUMBL_DATA_URL,
  link: from([error, authLink, responseLink.concat(httpLink)]),
  defaultOptions: { query: { fetchPolicy: "cache-first" } },
});

export default client;
