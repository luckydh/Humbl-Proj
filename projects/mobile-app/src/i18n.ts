import i18n from "i18next";
import { initReactI18next, useTranslation } from "react-i18next";
import { register } from "timeago.js";
import { es, pt_BR } from "timeago.js/lib/lang";
import HttpApi from "i18next-http-backend";
import LanguageDetector from "i18next-browser-languagedetector";
//import resources from "./locales/index";
// resources: resources,

export type TFnType = ReturnType<typeof useTranslation>["t"];

register("es", es);
register("pt", pt_BR);

i18n
  .use(HttpApi)
  .use(LanguageDetector)
  .use(initReactI18next) // passes i18n down to react-i18next
  .init({
    backend: {
      loadPath: "https://api.locize.app/f4e7a8d0-8642-44bb-ad02-ce1d5554b83c/latest/{{lng}}/{{ns}}",
      addPath: "https://api.locize.app/f4e7a8d0-8642-44bb-ad02-ce1d5554b83c/latest/add/{{lng}}/{{ns}}",
      allowMultiLoading: false,
    },
    //resources,
    //If we don't have a transalation and only have empty string, display fallback
    returnEmptyString: false,
    fallbackLng: "en",
    debug: false,
    ns: "translation",
    keySeparator: false, // we do not use keys in form messages.welcome
    interpolation: {
      escapeValue: false, // react already safes from xss
    },
  });

export default i18n;
