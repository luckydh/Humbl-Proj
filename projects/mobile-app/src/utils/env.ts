export type VersionObject = {
  major: string;
  minor: string;
  build: string;
};

export function buildVersionString(testingVersionObject?: VersionObject): string {
  const majorVersionNumber = testingVersionObject?.major ?? process.env.REACT_APP_VERSION_MAJOR;
  const minorVersionNumber = testingVersionObject?.minor ?? process.env.REACT_APP_VERSION_MINOR;
  const buildVersionNumber = testingVersionObject?.build ?? process.env.REACT_APP_CI_BUILD_NUMBER;

  const majorMinorString = `${majorVersionNumber}.${minorVersionNumber}`;

  return buildVersionNumber ? `${majorMinorString}.${buildVersionNumber}` : majorMinorString;
}
