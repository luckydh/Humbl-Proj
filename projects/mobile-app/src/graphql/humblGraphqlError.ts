import { ApolloError } from "@apollo/client";
import { GraphQLError } from "graphql";
import i18n from "i18n";

export type HumblErrorCode = keyof typeof errorCodesTranslationKeys;

export interface HumblError {
  humblErrorCode: HumblErrorCode;
}

export interface HumblGraphQLError extends GraphQLError {
  humblErrorCode?: HumblErrorCode;
}

const errorCodesTranslationKeys = {
  SERVER_ERROR: "crypto-wallet.buy.order-preview.error-modal.fallback-message",
  IDENTITY_MISMATCH: "crypto-wallet.buy.order-preview.error-modal.identity-mismatch",
  PHONE_MISMATCH_INVALID: "crypto-wallet.buy.order-preview.error-modal.phone-mismatch-invalid",
  TOO_MUCH_ABUSE_SUSPICIOUS: "crypto-wallet.buy.order-preview.error-modal.abuse-suspicious",
  AUTHORIZATION_FAILED: "crypto-wallet.buy.order-preview.error-modal.auth-failed",
  ORDER_EXPIRED: "crypto-wallet.buy.order-preview.error-modal.order-expired",
  DECLINED_DO_NOT_HONOR: "crypto-wallet.buy.order-preview.error-modal.declined-do-not-honor",
  INSUFFICIENT_FUNDS: "crypto-wallet.buy.order-preview.error-modal.insufficient-funds",
  RESTRICTED: "crypto-wallet.buy.order-preview.error-modal.restricted",
  CARD_ISSUER_DECLINED: "crypto-wallet.buy.order-preview.error-modal.card-issuer-declined",
  LIMIT_REACHED: "crypto-wallet.buy.order-preview.error-modal.limit-reached",
  VALIDATION_ERROR: "crypto-wallet.buy.order-preview.error-modal.validation-error",
};

const fallback = errorCodesTranslationKeys.SERVER_ERROR;

export function getPotentialHumblError(graphQLErrors: ReadonlyArray<HumblGraphQLError>): HumblError | undefined {
  // This can sometimes trigger with 0 graphQLErrors hence this guard.
  const humblErrorCode = graphQLErrors[0]?.humblErrorCode;
  return humblErrorCode && { humblErrorCode };
}

export function getErrorMessage(error?: ApolloError) {
  if (!error) {
    return;
  }
  const humblError = getPotentialHumblError(error.graphQLErrors);
  return i18n.t(mapErrorCodeToTranslationKey(humblError?.humblErrorCode));
}

export function mapErrorCodeToTranslationKey(errorCode: HumblErrorCode = "SERVER_ERROR") {
  return errorCodesTranslationKeys[errorCode] ?? fallback;
}
