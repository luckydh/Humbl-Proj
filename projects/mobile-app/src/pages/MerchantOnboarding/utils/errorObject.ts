import { TFnType } from "i18n";
import { LanguageMessages } from "joi";
import { useTranslation } from "react-i18next";

export function useGenericErrorMessages(): LanguageMessages {
  const { t } = useTranslation();
  return {
    "any.required": t("input.error.required"),
    "string.empty": t("input.error.string-empty"),
    "string.max": t("input.error.string-too-long"),
    "string.min": t("input.error.string-too-short"),
    "number.base": t("input.error.number-required"),
    "date.format": t("input.error.date-format"),
    "string.email": t("input.error.email-format"),
    "date.base": t("input.error.date-invalid"),
    "date.max": t("input.error.too-young"),
    "date.min": t("input.error.too-old"),
  };
}

export function makeGenericErrorMessages(t: TFnType): LanguageMessages {
  return {
    "any.required": t("input.error.required"),
    "string.empty": t("input.error.string-empty"),
    "string.max": t("input.error.string-too-long"),
    "string.min": t("input.error.string-too-short"),
    "number.base": t("input.error.number-required"),
    "date.format": t("input.error.date-format"),
    "string.email": t("input.error.email-format"),
    "date.base": t("input.error.date-invalid"),
    "date.max": t("input.error.too-young"),
    "date.min": t("input.error.too-old"),
  };
}
