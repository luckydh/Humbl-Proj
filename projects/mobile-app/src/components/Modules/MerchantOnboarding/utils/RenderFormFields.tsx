import React from "react";
import { Control } from "react-hook-form";
import Input from "components/Input/Input";
import { Message } from "components/Message/Message";
import { Checkbox } from "components/Checkbox/Checkbox";
import { FieldDefinitions } from "./getSchema";
import { Select } from "components/Select/Select";
import Label from "components/Label/Label";
import { IonIcon } from "@ionic/react";
import { InfoIcon } from "assets/icons";

interface Props {
  fields: FieldDefinitions[];
  register: any;
  control: Control;
  errors: any;
  prepend?: string;
}

export const RenderFormFields = ({ fields, register, errors, control, prepend = "" }: Props) => {
  const final = fields.map((field) => {
    const fieldName = `${prepend}${field.name}`;
    if (field.component) {
      const Component = field.component;
      return (
        <div key={fieldName}>
          <Component register={register} errors={errors} field={field} control={control} {...field.componentProps} />
        </div>
      );
    }

    if (field.type === "checkbox") {
      return (
        <div key={fieldName} className="flex align-middle items-center justify-between mx-6">
          <span>{field.label}</span>
          <Checkbox size="large" name={fieldName} control={control} defaultValue={false} />
        </div>
      );
    }

    if (field?.options && field.options.length > 0) {
      return (
        <div key={fieldName} className="mb-4">
          <Label>{field.label}</Label>
          <Select
            options={field.options}
            name={fieldName}
            register={register}
            placeholder={field.label}
            hasError={errors[fieldName]}
          />
          {errors[fieldName]?.message && (
            <div className="my-2">
              <Message variant="error">{errors[fieldName].message}</Message>
            </div>
          )}
        </div>
      );
    }

    return (
      <div key={fieldName}>
        <Input
          label={field.label}
          name={fieldName}
          register={register}
          placeholder={field.label}
          hasError={errors[fieldName]}
          isSecure={field.isSecure}
        />
        {field.infoText && (
          <div className="flex items-center mt-2 mb-4 ml-2">
            <IonIcon className="text-xs mr-2" icon={InfoIcon} />
            <p className="text-white text-xs opacity-80 ">{field.infoText}</p>
          </div>
        )}
        {errors[fieldName]?.message && (
          <div className="my-2">
            <Message variant="error">{errors[fieldName].message}</Message>
          </div>
        )}
      </div>
    );
  });
  return <>{final}</>;
};
