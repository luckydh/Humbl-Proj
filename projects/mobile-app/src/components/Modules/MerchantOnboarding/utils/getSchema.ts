import Joi, { AnySchema, ObjectSchema } from "joi";
import React from "react";
import getOptions, { SelectOption } from "pages/MerchantOnboarding/utils/getOptions";
import { Countries } from "utils/Countries";

export type FieldDefinitions = {
  country: Countries[];
  name: string;
  label?: string;
  schema?: AnySchema;
  component?: React.FunctionComponent<any>;
  componentProps?: Record<string, any>;
  options?: SelectOption[];
  componentSchema?: SchemaRecord[];
  infoText?: string;
  type?: "input" | "select" | "checkbox";
  isSecure?: boolean;
};

export type ComponentSchemaRecord = { [key: string]: Joi.ObjectSchema }[];
export type SchemaRecord = { [key: string]: Joi.AnySchema };

export type FormComponentProps = {
  register?: any;
  field?: FieldDefinitions;
  errors?: any;
};

export const getSchema = (fields: FieldDefinitions[], country: Countries) => {
  const filtered = fields.filter((elem) => elem.country.includes(country));
  let schema: ObjectSchema = Joi.object({
    test: Joi.string(),
  }).unknown();
  filtered.forEach((elem) => {
    if (elem.component) {
      elem.componentSchema?.forEach((rule) => {
        schema = schema.concat(Joi.object(rule).unknown());
      });
      return;
    }

    if (elem.schema) {
      /**
       * TODO: This was partially working, but currently removed
       * any dot named fields in schema. React Hook Form supports nested structure
       * so that form results can be stored as objects and arrays, but the JOI
       * format has to be different to match the object/array fields.
       * Paused work on this as it won't be called
       */
      const isNested = elem.name.split(".");
      if (isNested.length > 1) {
        const newKey: SchemaRecord = {};
        newKey[elem.name] = Joi.object({ [isNested[0]]: elem.schema });
        // Possible solution below, but untested due to time.
        // newKey[elem.name] = Joi.object().keys({ [isNested[0]]: { [isNested[0]]: elem.schema } });
        schema = schema.concat(Joi.object(newKey));
      } else {
        const newKey: SchemaRecord = {};
        newKey[elem.name] = elem.schema;
        schema = schema.concat(Joi.object(newKey));
      }

      if (
        elem.name === "addressState" ||
        elem.name === "individualAddressState" ||
        elem.name === "owner_0_addressState"
      ) {
        elem.options = getOptions(country);
      }
    }
  });

  return {
    fields: filtered,
    schema,
  };
};
