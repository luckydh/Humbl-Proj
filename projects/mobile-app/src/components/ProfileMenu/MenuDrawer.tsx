import { IonIcon, IonMenu, IonMenuToggle } from "@ionic/react";
import Button from "components/Button/Button";
import React, { useState } from "react";
import { HumblLogoResizeable } from "assets/svgs/HumblLogoResizeable";
import { profileIcon, chevronForwardIcon } from "assets/icons";
import { Link, useHistory } from "react-router-dom";
import { useTranslation } from "react-i18next";
import "./styles.scss";
import QRModal from "components/QR/QRModal";
import { useGetCurrentAccount } from "hooks/useGetCurrentAccount";
import { Share } from "@capacitor/share";
import { trackEvent } from "utils/analytics/Segment";
import EVENTS from "utils/analytics/AnalyticEvents";
import { FeedbackTrigger } from "pages/CryptoWallet/FeedbackTrigger";
import { useMenuItems } from "./useMenuItems";
import { AccountType } from "../../generated/graphql";
import { VersionObject, buildVersionString } from "utils/env";

export type MenuOption = {
  text: string;
  link: string | { pathname: string };
  target?: string;
  isSingOut?: boolean;
  action?: string;
  icon?: string;
  hide?: boolean;
};

type MenuDrawerProps = {
  accountProp?: AccountType;
  /** Only for testing  */
  testingVersionObject?: VersionObject;
};

const MenuDrawer: React.FC<MenuDrawerProps> = ({ accountProp, testingVersionObject }) => {
  const [showQRModal, setShowQRModal] = useState(false);
  // const account = useCurrentAccount();
  let account: AccountType;
  const { currentAccount: accountReq } = useGetCurrentAccount();

  // This is primarily used for dependency injection for tests.
  if (accountProp) {
    account = { ...accountProp };
  } else {
    account = { ...(accountReq as AccountType) };
  }

  const actions: { [key: string]: () => void } = {
    setShowQRModal: () => setShowQRModal(true),
  };
  const history = useHistory();
  const { t } = useTranslation();

  const versionString = buildVersionString(testingVersionObject);
  const options = useMenuItems(account);

  return (
    <div className="safe-area-top">
      <IonMenu side="start" menuId="menu" contentId="main_content" type="overlay" swipeGesture={false}>
        <div className="bg-blue h-full">
          <div className="w-full flex relative items-center">
            <IonMenuToggle className="my-3 sm:my-6 absolute left-5 top-0 ">
              <IonIcon icon="close" className="text-3xl text-white" />
            </IonMenuToggle>
            <h2 className="flex-1 text-xl text-white my-3 sm:m-6 text-center">Menu</h2>
          </div>
          <div className="px-6">
            <IonMenuToggle>
              {options.map(({ hide, link, text, action: actionKey, icon, target }) => {
                if (hide) return null;
                return (
                  <Link
                    key={text}
                    replace={false}
                    to={link}
                    className="justify-between text-white group flex items-center px-0 py-2 text-lg font-medium rounded-md"
                    target={target || "_self"}
                    onClick={(e) => {
                      if (actionKey) {
                        e.preventDefault();
                        const action = actions[actionKey];
                        action && action();
                      }
                      if (link === "/merchantcreate") {
                        trackEvent(EVENTS.MERCHANT_CREATION_INITIATED, { previousPage: history.location.pathname });
                      }
                    }}>
                    <div className="flex items-center align-middle">
                      <IonIcon className="text-white mr-4 h-6 w-6" icon={icon} />
                      <h2 className="text-white">{t(text)}</h2>
                    </div>
                    <IonIcon className="text-xl" icon={chevronForwardIcon} />
                  </Link>
                );
              })}
              {account?.hasMultipleAccounts && (
                <button
                  type="button"
                  className="w-full justify-between text-white group flex items-center px-0 py-2 text-lg font-medium rounded-md"
                  onClick={() => {
                    history.push("/switch-accounts");
                  }}>
                  <div className="flex items-center align-middle">
                    <IonIcon className="text-white mr-4 h-6 w-6" icon={profileIcon} />
                    <h2 className="text-lg">{t("pages-switch-account.title.switch-accounts")}</h2>
                  </div>
                  <IonIcon className="text-xl text-white" icon={chevronForwardIcon} />
                </button>
              )}
              <FeedbackTrigger>
                <div className="w-full flex items-center justify-center text-center pt-8">
                  {t("pull_out_menu_version")} {versionString}
                </div>
              </FeedbackTrigger>
              <div className="flex justify-center mt-3 sm:mt-12 pb-6 w-full">
                <Button
                  onClick={async () => {
                    await Share.share({
                      title: t("component-share-widget.title"),
                      text: t("component-share-widget.text"),
                      url: `https://www.humblpay.com/get-humbl/?invitedby=${account?.id}`,
                      dialogTitle: t("component-share-widget.dialogTitle"),
                    });
                  }}>
                  <div className="flex">
                    <span className="inline-block text-xl font-semibold">
                      {t("component-profile-menu.button.invite-friends-to-humbl")}
                    </span>
                    <span className="inline-block m-auto ml-1.5">
                      <div className="w-16">
                        <HumblLogoResizeable />
                      </div>
                    </span>
                  </div>
                </Button>
              </div>
            </IonMenuToggle>
          </div>
        </div>
      </IonMenu>
      <QRModal setShowQRModal={setShowQRModal} showQRModal={showQRModal} />
    </div>
  );
};

export default MenuDrawer;
