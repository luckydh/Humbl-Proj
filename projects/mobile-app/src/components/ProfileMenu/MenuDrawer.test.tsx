import React from "react";
import { StaticRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import MenuDrawer from "./MenuDrawer";

// Disable suspense on menu drawer
import i18n from "../../i18n";
import { AccountType } from "generated/graphql";
import { MockedProvider } from "@apollo/client/testing";
import { VersionObject } from "utils/env";

if (i18n.options) {
  i18n.options.react = {
    useSuspense: false,
  };
}

const testingVersionObject: VersionObject = { major: "2", minor: "0", build: "2110" };

it("renders menu drawer correctly", async () => {
  const accountMock = {
    city: "randomUser.location.city",
    country: "AnyCountry",
    displayName: "Foo Man",
    id: "1234",
    phone: "1234567891",
    image: "https://place-hold.it/300.png",
    userName: "randomUser.login@username.com",
    isMerchant: false,
    hasMultipleAccounts: true,
  };
  const tree = renderer
    .create(
      <StaticRouter>
        <MockedProvider>
          <MenuDrawer testingVersionObject={testingVersionObject} accountProp={accountMock as AccountType} />
        </MockedProvider>
      </StaticRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
