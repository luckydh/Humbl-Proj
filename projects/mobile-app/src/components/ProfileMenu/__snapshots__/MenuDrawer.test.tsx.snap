// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders menu drawer correctly 1`] = `
<div
  className="safe-area-top"
>
  <ion-menu
    content-id="main_content"
    menu-id="menu"
    side="start"
    swipe-gesture={false}
    type="overlay"
  >
    <div
      className="bg-blue h-full"
    >
      <div
        className="w-full flex relative items-center"
      >
        <ion-menu-toggle>
          <ion-icon
            icon="close"
          />
        </ion-menu-toggle>
        <h2
          className="flex-1 text-xl text-white my-3 sm:m-6 text-center"
        >
          Menu
        </h2>
      </div>
      <div
        className="px-6"
      >
        <ion-menu-toggle>
          <a
            className="justify-between text-white group flex items-center px-0 py-2 text-lg font-medium rounded-md"
            href="/update-profile"
            onClick={[Function]}
            target="_self"
          >
            <div
              className="flex items-center align-middle"
            >
              <ion-icon
                icon="ProfileIcon.svg"
              />
              <h2
                className="text-white"
              >
                component-profile-menu.text.my-profile
              </h2>
            </div>
            <ion-icon
              icon="ChevronForwardIcon.svg"
            />
          </a>
          <a
            className="justify-between text-white group flex items-center px-0 py-2 text-lg font-medium rounded-md"
            href="/"
            onClick={[Function]}
            target="_self"
          >
            <div
              className="flex items-center align-middle"
            >
              <ion-icon
                icon="QrIcon.svg"
              />
              <h2
                className="text-white"
              >
                component-profile-menu.text.my-qr-code
              </h2>
            </div>
            <ion-icon
              icon="ChevronForwardIcon.svg"
            />
          </a>
          <a
            className="justify-between text-white group flex items-center px-0 py-2 text-lg font-medium rounded-md"
            href="/payment-methods"
            onClick={[Function]}
            target="_self"
          >
            <div
              className="flex items-center align-middle"
            >
              <ion-icon
                icon="CardIcon.svg"
              />
              <h2
                className="text-white"
              >
                component-profile-menu.text.payment-methods
              </h2>
            </div>
            <ion-icon
              icon="ChevronForwardIcon.svg"
            />
          </a>
          <a
            className="justify-between text-white group flex items-center px-0 py-2 text-lg font-medium rounded-md"
            href="/transactions"
            onClick={[Function]}
            target="_self"
          >
            <div
              className="flex items-center align-middle"
            >
              <ion-icon
                icon="MoneyIcon.svg"
              />
              <h2
                className="text-white"
              >
                component-profile-menu.text.transactions
              </h2>
            </div>
            <ion-icon
              icon="ChevronForwardIcon.svg"
            />
          </a>
          <a
            className="justify-between text-white group flex items-center px-0 py-2 text-lg font-medium rounded-md"
            href="/crypto-wallet"
            onClick={[Function]}
            target="_self"
          >
            <div
              className="flex items-center align-middle"
            >
              <ion-icon
                icon="CryptoWallet.svg"
              />
              <h2
                className="text-white"
              >
                component-profile-menu.text.crypto-wallet
              </h2>
            </div>
            <ion-icon
              icon="ChevronForwardIcon.svg"
            />
          </a>
          <a
            className="justify-between text-white group flex items-center px-0 py-2 text-lg font-medium rounded-md"
            href="/offers/blocks"
            onClick={[Function]}
            target="_self"
          >
            <div
              className="flex items-center align-middle"
            >
              <ion-icon
                icon="icon-invest.svg"
              />
              <h2
                className="text-white"
              >
                component-profile-menu.text.blocks-etx
              </h2>
            </div>
            <ion-icon
              icon="ChevronForwardIcon.svg"
            />
          </a>
          <a
            className="justify-between text-white group flex items-center px-0 py-2 text-lg font-medium rounded-md"
            href="/merchantcreate"
            onClick={[Function]}
            target="_self"
          >
            <div
              className="flex items-center align-middle"
            >
              <ion-icon
                icon="CartIcon.svg"
              />
              <h2
                className="text-white"
              >
                component-profile-menu.text.become-merchant
              </h2>
            </div>
            <ion-icon
              icon="ChevronForwardIcon.svg"
            />
          </a>
          <a
            className="justify-between text-white group flex items-center px-0 py-2 text-lg font-medium rounded-md"
            href="/settings"
            onClick={[Function]}
            target="_self"
          >
            <div
              className="flex items-center align-middle"
            >
              <ion-icon
                icon="GearIcon.svg"
              />
              <h2
                className="text-white"
              >
                component-profile-menu.text.settings
              </h2>
            </div>
            <ion-icon
              icon="ChevronForwardIcon.svg"
            />
          </a>
          <button
            className="w-full justify-between text-white group flex items-center px-0 py-2 text-lg font-medium rounded-md"
            onClick={[Function]}
            type="button"
          >
            <div
              className="flex items-center align-middle"
            >
              <ion-icon
                icon="ProfileIcon.svg"
              />
              <h2
                className="text-lg"
              >
                pages-switch-account.title.switch-accounts
              </h2>
            </div>
            <ion-icon
              icon="ChevronForwardIcon.svg"
            />
          </button>
          <div
            onClick={[Function]}
          >
            <div
              className="w-full flex items-center justify-center text-center pt-8"
            >
              pull_out_menu_version
               
              2.0.2110
            </div>
          </div>
          <div
            className="flex justify-center mt-3 sm:mt-12 pb-6 w-full"
          >
            <button
              className="text-white select-none outline-none flex justify-center items-center justify-center font-medium w-full text-white py-2 px-4 bg-blue-dark rounded-md active:bg-blue-light active:shadow-button-active active:text-blue-dark transition-all py-3 px-6 text-md sm:leading-none"
              onClick={[Function]}
              style={
                Object {
                  "minWidth": 160,
                }
              }
              type="button"
            >
              <div
                className="flex"
              >
                <span
                  className="inline-block text-xl font-semibold"
                >
                  component-profile-menu.button.invite-friends-to-humbl
                </span>
                <span
                  className="inline-block m-auto ml-1.5"
                >
                  <div
                    className="w-16"
                  >
                    <svg
                      className="block w-full"
                      viewBox="0 0 143 31"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <g
                        fill="#FFF"
                        fillRule="nonzero"
                      >
                        <path
                          d="M18.418 1v11.484H7.151V1H0v29.794h7.151V18.335h11.267v12.459h7.259V1zM37.594 1v17.118a7.15 7.15 0 001.3 4.55 4.659 4.659 0 003.9 1.517 5.092 5.092 0 003.9-1.517 6.934 6.934 0 001.3-4.55V1h7.151v17.118c.09 2.45-.47 4.88-1.625 7.042a11.376 11.376 0 01-4.442 4.334 13.543 13.543 0 01-6.392 1.406c-2.18.06-4.339-.423-6.284-1.408a10.4 10.4 0 01-4.334-4.334 14.518 14.518 0 01-1.517-7.148V1h7.043zM93.39 1.217v29.577h-7.151V12.376l-6.392 18.527h-6.175L67.28 12.376v18.527h-7.151V1.217h8.776l8.017 20.585 7.907-20.585zM120.475 18.118a6.934 6.934 0 011.625 4.55 7.476 7.476 0 01-2.492 5.959 11.159 11.159 0 01-7.259 2.167H98.482V.998h13.543c2.5-.159 4.98.528 7.042 1.95a6.934 6.934 0 012.492 5.634 6.609 6.609 0 01-1.408 4.442 6.934 6.934 0 01-3.792 2.384 8.884 8.884 0 014.117 2.709m-14.952-4.875h4.767c2.492 0 3.792-1.083 3.792-3.25s-1.3-3.25-3.792-3.25h-4.658l-.109 6.5zm9.209 8.451a2.817 2.817 0 00-1.083-2.492 4.225 4.225 0 00-3.034-.867h-5.092v6.609h5.092c2.709 0 4.117-1.083 4.117-3.25M133.044 25.268h9.642v5.525h-16.793V1h7.151zM140.519 1a2.167 2.167 0 110 4.334 2.167 2.167 0 010-4.334m0 4.117a1.95 1.95 0 10-1.95-1.95 1.95 1.95 0 001.95 1.95m-.758-3.25h.975c.542 0 .867.217.867.758a.65.65 0 01-.65.65l.758 1.192h-.217l-.758-1.192h-.65v1.192h-.325v-2.6zm.217 1.192h.65c.433 0 .65-.108.65-.433s-.217-.542-.65-.542h-.65v.975z"
                        />
                      </g>
                    </svg>
                  </div>
                </span>
              </div>
            </button>
          </div>
        </ion-menu-toggle>
      </div>
    </div>
  </ion-menu>
</div>
`;
